src/tools/opensc-explorer.c

```diff
@@ -1469,6 +1478,92 @@ static int do_get(int argc, char **argv)
 	return -err;
 }
 
+static int do_get_record(int argc, char **argv)
+{
+	u8 buf[SC_MAX_EXT_APDU_RESP_SIZE];
+	int r, err = 1;
+	size_t rec, count = 0;
+	sc_path_t path;
+	sc_file_t *file = NULL;
+	char *filename;
+	FILE *outf = NULL;
+
+	if (argc < 2 || argc > 3)
+		return usage(do_get);
+	if (arg_to_path(argv[0], &path, 0) != 0)
+		return usage(do_get);
+	rec = strtoul(argv[1], NULL, 10);
+
+	filename = (argc == 3) ? argv[2] : path_to_filename(&path, '_', rec);
+	outf = (strcmp(filename, "-") == 0)
+		? stdout
+		: fopen(filename, "wb");
```

Comment:
`setmode(fileno(stdout), O_BINARY)`?

Reply:
@frankmorgner Thanks for the hint. I will add it also to the other functions writing potentially binary data to stdout.